hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigGeneratorUtil.java

```diff
@@ -242,6 +244,23 @@ public static String pnCounterConfigGenerator(PNCounterConfig subConfig, boolean
         );
     }
 
+    public static String tcpIpConfigGenerator(
+            TcpIpConfig subConfig, boolean configIsXml, int indent, boolean advancedNetwork
+    ) {
+        return configGenerator(subConfig, configIsXml, indent,
+                TCP_IP.getName(),
+                (config, tcpIpConfig) -> {
+                    if (advancedNetwork) {
+                        config.getAdvancedNetworkConfig().getJoin().setTcpIpConfig(tcpIpConfig);
+                    } else {
+                        config.getNetworkConfig().getJoin().setTcpIpConfig(tcpIpConfig);
+                    }
```

Comment:
I believe here it's better to not use`ConfigAccessor.getActiveMemberNetworkConfig(Config config)` since `config` comes from the current config, and the `advancedNetwork` flag is come from the updating config.